api/include/opentelemetry/nostd/type_traits.h

```diff
@@ -133,7 +133,7 @@ using is_nothrow_swappable = detail::swappable::is_nothrow_swappable<is_swappabl
  *  std::is_trivialy_copy_assignable
  *  std::is_trivialy_move_assignable
  */
-#ifdef OPENTELEMETRY_TRIVIALITY_TYPE_TRAITS
+#if defined(OPENTELEMETRY_STL_VERSION) && OPENTELEMETRY_STL_VERSION >= 2011
```

Comment:
The fallback implementation relies on GCC's API, making it exclusive to GCC. It should be updated to support all compilers.

Reply:
This is a dead code, opentelemetry does not support cxx03 standard / libstdc++ 5.

Reply:
@georgthegreat 

In theory, yes, this is dead code.
In practice, no, this code is not totally dead now, there are historical reasons to maintain this part working.

What we need here is a fix that solves the trivial cases of building with recent compilers, using `std:: type` traits when they are available, instead of using the alternate `nostd::` implementation.

The issue is probably that we need to do this independently of `WITH_STL` : that is, test `__cplusplus` instead of `OPENTELEMETRY_STL_VERSION`.

Now, for old cases when `std::` classes are not available, like this old work around for gcc 4.8, the code needs to implement the work around as before.

Reply:
I hope I understood you correctly.

Reply:
`#if !defined(OPENTELEMETRY_HAVE_STD_TYPE_TRAITS)` can be removed now.